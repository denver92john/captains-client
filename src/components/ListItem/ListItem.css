.list-item {
    border-bottom: 1px solid black;
	padding: 7px;
	display: flex;
	justify-content: space-between;
}

.list-item:last-of-type {
    border-bottom: none;
}

.list-item-link {
    color: var(--color1);
    font-weight: 500;
    font-size: 20px;
}

.list-item-p {
    font-size: 20px;
}