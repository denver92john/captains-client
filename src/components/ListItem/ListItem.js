import React from 'react';
import {Link} from 'react-router-dom';
import FormItem from '../FormItem/FormItem';
import '../List/List.css';

const ifList = (item, link) => (
    (item.list_name)
        ? <Link to={link}>{item.list_name}</Link>
        : <p>{item.item_name}</p>
)

function ListItem(props) {
    const {item, link, onPatch, onDelete} = props;
    const content = ifList(item, link);

    return (
        <li className="named-list-item">
            {content}
            <FormItem 
                item={item} 
                onPatch={onPatch} 
                onDelete={onDelete} 
            />
        </li>
    );
}

export default ListItem;