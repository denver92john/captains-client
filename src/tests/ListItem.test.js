import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import ListItem from '../components/ListItem/ListItem';

describe('Listitem component', () => {
    const listProp = {
        id: 2,
        list_name: "Turkey Bowl",
        user_id: 1
    }
    const itemProp = {
        id: 1,
        item_name: "Joe",
        list_id: 2
    }
    const link = "/list/2/list-items"

    it('renders list without errors', () => {
        const ul = document.createElement('ul');
        ReactDOM.render(
            <BrowserRouter>
                <ListItem item={listProp} link={link} />
            </BrowserRouter>,
            ul);
        ReactDOM.unmountComponentAtNode(ul);
    });

    it('renders item without errors', () => {
        const ul = document.createElement('ul');
        ReactDOM.render(
            <BrowserRouter>
                <ListItem item={itemProp} />
            </BrowserRouter>,
            ul);
        ReactDOM.unmountComponentAtNode(ul);
    });

    it('renders UI as expected', () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <ListItem item={itemProp} />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})