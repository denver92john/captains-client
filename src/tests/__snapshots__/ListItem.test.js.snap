// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listitem component renders UI as expected 1`] = `
<li
  className="list-item"
>
  <p
    className="list-item-p"
  >
    Joe
  </p>
  <div>
    <button
      className="formitem-button"
      onClick={[Function]}
      type="button"
    />
    <button
      className="formitem-button"
      onClick={[Function]}
      type="button"
    />
  </div>
</li>
`;
